app/src/userpages/modules/userPageStreams/actions.js

```diff
@@ -162,9 +139,29 @@ export const clearStreamsList = () => (dispatch: Function) => (
     dispatch(clearStreamsListAction())
 )
 
+const mapStreamFields = (stream: Stream): Stream => {
+    const { config } = stream
+
+    if (!config || !config.fields) {
+        return stream
+    }
+
+    return {
+        ...stream,
+        config: {
+            ...config,
+            fields: (config.fields || []).map((field) => ({
+                ...field,
+                id: field.id ? field.id : uuid(),
```

Comment:
ids for fields are generated now here on stream load instead of the field config component mount.